connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -72,115 +41,42 @@
+    ) {
+        super(numBrokers, brokerProps, maskExitProcedures, clientProps);
         this.workerProps = workerProps;
         this.connectClusterName = name;
-        this.numBrokers = numBrokers;
-        this.kafkaCluster = new EmbeddedKafkaCluster(numBrokers, brokerProps, additionalKafkaClusterClientConfigs);
         this.connectCluster = new LinkedHashSet<>();
-        this.httpClient = new HttpClient();
         this.numInitialWorkers = numWorkers;
-        this.maskExitProcedures = maskExitProcedures;
         // leaving non-configurable for now
         this.workerNamePrefix = DEFAULT_WORKER_NAME_PREFIX;
-        this.assertions = new EmbeddedConnectClusterAssertions(this);
-    }
-
-    /**
-     * A more graceful way to handle abnormal exit of services in integration tests.
-     */
-    public Exit.Procedure exitProcedure = (code, message) -> {
-        if (code != 0) {
-            String exitMessage = "Abrupt service exit with code " + code + " and message " + message;
-            log.warn(exitMessage);
-            throw new UngracefulShutdownException(exitMessage);
-        }
-    };
-
-    /**
-     * A more graceful way to handle abnormal halt of services in integration tests.
-     */
-    public Exit.Procedure haltProcedure = (code, message) -> {
-        if (code != 0) {
-            String haltMessage = "Abrupt service halt with code " + code + " and message " + message;
-            log.warn(haltMessage);
-            throw new UngracefulShutdownException(haltMessage);
-        }
-    };
-
-    /**
-     * Start the connect cluster and the embedded Kafka and Zookeeper cluster.
-     */
-    public void start() {
-        if (maskExitProcedures) {
-            Exit.setExitProcedure(exitProcedure);
-            Exit.setHaltProcedure(haltProcedure);
-        }
-        kafkaCluster.start();
-        startConnect();
-        try {
-            httpClient.start();
-        } catch (Exception e) {
-            throw new ConnectException("Failed to start HTTP client", e);
-        }
-    }
-
-    /**
-     * Stop the connect cluster and the embedded Kafka and Zookeeper cluster.
-     * Clean up any temp directories created locally.
-     *
-     * @throws RuntimeException if Kafka brokers fail to stop
-     */
-    public void stop() {
-        Utils.closeQuietly(httpClient::stop, "HTTP client for embedded Connect cluster");
-        connectCluster.forEach(this::stopWorker);
-        try {
-            kafkaCluster.stop();
-        } catch (UngracefulShutdownException e) {
-            log.warn("Kafka did not shutdown gracefully");
-        } catch (Exception e) {
-            log.error("Could not stop kafka", e);
-            throw new RuntimeException("Could not stop brokers", e);
-        } finally {
-            if (maskExitProcedures) {
-                Exit.resetExitProcedure();
-                Exit.resetHaltProcedure();
-            }
-            Plugins.compareAndSwapLoaders(originalClassLoader);
-        }
```

Comment:
This is all migrated directly (i.e., copy+pasted) to the `EmbeddedConnect` class.